File: Behavioural-Patterns/ChainOfResponsibility/C#/ChainOfResponsibility/Interfaces/BaseHandler.cs
public abstract class BaseHandler : IRequestHandler {
    protected IRequestHandler? next;

    public IRequestHandler SetNext(IRequestHandler next){
        this.next = next;
        return next;
    }

    public void Handle(Request request){
        bool continueChain = processRequest(request);

        if(continueChain && next != null){
            next.Handle(request);
        }
    }

    protected abstract bool processRequest(Request request);
}

File: Behavioural-Patterns/ChainOfResponsibility/Java/Interfaces/BaseHandler.java
package Interfaces;

import Implementations.Request;

public abstract class BaseHandler implements IRequestHandler {
    protected IRequestHandler next;

    @Override
    public IRequestHandler setNext(IRequestHandler next){
        this.next = next;
        return next;
    }

    @Override
    public final void handle(Request request){
        boolean continueChain = processRequest(request);

        if(continueChain && next != null){
            next.handle(request);
        }
    }

    protected abstract boolean processRequest(Request request);
}
